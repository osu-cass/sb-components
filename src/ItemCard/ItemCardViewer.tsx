import * as React from "react";
import { AboutItemModel } from "../AboutItem/AboutItemModels";
import { ItemViewerFrame, Rubric, AboutThisItemDetail } from "../index";
import { Tabs, ItemTabs } from "../PageTabs/ItemTabs";

export interface ItemCardViewerProps {
  item?: AboutItemModel;
}

export interface State {
  selectedTab: Tabs;
}

export class ItemCardViewer extends React.Component<
  ItemCardViewerProps,
  State
> {
  constructor(props: ItemCardViewerProps) {
    super(props);
    this.state = {
      selectedTab: "viewer"
    };
  }

  onTabChange(tab: Tabs) {
    this.setState({
      selectedTab: tab
    });
  }

  renderViewer(url: string) {
    return (
      <div className="item-content">
        <ItemViewerFrame url={url} />
      </div>
    );
  }

  renderRubric() {
    if (this.props.item) {
      const rubrics = this.props.item.rubrics.map((ru, i) => (
        <Rubric {...ru} key={String(i)} />
      ));
      return <div className="item-content">{rubrics}</div>;
    }
  }

  renderInformation() {
    if (this.props.item) {
      const aboutItem = this.props.item;
      return (
        <div className="item-content">
          <div>
            <AboutThisItemDetail {...aboutItem} />
          </div>
        </div>
      );
    }
  }

  renderChosen() {
    let selectedTab = null;
    if (this.props.item) {
      const selectedTab = this.state.selectedTab;
      const itemCard = this.props.item.itemCardViewModel;

      let resultElement: JSX.Element[] | JSX.Element | undefined;
      if (selectedTab == "viewer") {
        const url =
          "http://ivs.smarterbalanced.org/items?ids=" +
          itemCard.bankKey.toString() +
          "-" +
          itemCard.itemKey.toString();
        resultElement = <div> {this.renderViewer(url)} </div>;
      } else if (selectedTab == "rubric") {
        resultElement = <div> {this.renderRubric()} </div>;
      } else if (selectedTab == "information") {
        resultElement = <div> {this.renderInformation()}</div>;
      }
      return resultElement;
    }
  }
  render() {
    return (
      <div className="item-card">
        <ItemTabs
          changedTab={tab => this.onTabChange(tab)}
          selectedTab={this.state.selectedTab}
        />
        {this.renderChosen()}
      </div>
    );
  }
}
