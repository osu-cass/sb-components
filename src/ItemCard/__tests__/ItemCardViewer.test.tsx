import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemCardViewer } from '../ItemCardViewer';
import * as TestUtils from 'react-dom/test-utils';
import { shallow, mount, render } from 'enzyme';

import { ItemCardModel, RubricModel, AboutItemModel } from '../../';


describe("ItemCardViewer", () => {
    const itemCardViewModel: ItemCardModel = {
        bankKey: 1,
        itemKey: 123,
        title: "test_title",
        grade: 1,
        gradeLabel: "1",
        subjectCode: "Math",
        subjectLabel: "MTH",
        claimCode: "claim_1234",
        claimLabel: "claim_math",
        targetShortName: "something",
        interactionTypeCode: "type01",
        interactionTypeLabel: "type01label",
        isPerformanceItem: false,
        targetHash: 323
    }

    const rubrics: RubricModel[] = []

    const item: AboutItemModel = {
        rubrics,
        itemCardViewModel,
        depthOfKnowledge: "depthOfKnowledge",
        targetDescription: "targetDescription",
        commonCoreStandardsDescription: "commonCoreStandardsDescription",
        educationalDifficulty: "educationalDifficulty",
        evidenceStatement: "evidenceStatement"
    }

    it("renders viewer tab", () => {
        let wrapper = shallow(<ItemCardViewer item={item} />);
        wrapper.setState({ selectedTab: "viewer" });
        expect(wrapper).toMatchSnapshot();
    })

    it("renders rubric tab", () => {
        let wrapper = shallow(<ItemCardViewer item={item} />);
        wrapper.setState({ selectedTab: "rubric" });
        expect(wrapper).toMatchSnapshot();
    })

    it("renders information tab", () => {
        let wrapper = shallow(<ItemCardViewer item={item} />);
        wrapper.setState({ selectedTab: "information" });
        expect(wrapper).toMatchSnapshot();
    })
})


