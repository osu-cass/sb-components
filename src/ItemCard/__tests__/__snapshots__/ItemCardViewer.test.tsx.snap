// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCardViewer renders information tab 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemCardViewer
    item={
      Object {
        "commonCoreStandardsDescription": "commonCoreStandardsDescription",
        "depthOfKnowledge": "depthOfKnowledge",
        "educationalDifficulty": "educationalDifficulty",
        "evidenceStatement": "evidenceStatement",
        "itemCardViewModel": Object {
          "bankKey": 1,
          "claimCode": "claim_1234",
          "claimLabel": "claim_math",
          "grade": 1,
          "gradeLabel": "1",
          "interactionTypeCode": "type01",
          "interactionTypeLabel": "type01label",
          "isPerformanceItem": false,
          "itemKey": 123,
          "subjectCode": "Math",
          "subjectLabel": "MTH",
          "targetHash": 323,
          "targetShortName": "something",
          "title": "test_title",
        },
        "rubrics": Array [],
        "targetDescription": "targetDescription",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ItemTabs
          changedTab={[Function]}
          selectedTab="information"
        />,
        <div>
           
          <div
            className="item-content"
          >
            <div>
              <AboutThisItemDetail
                commonCoreStandardsDescription="commonCoreStandardsDescription"
                depthOfKnowledge="depthOfKnowledge"
                educationalDifficulty="educationalDifficulty"
                evidenceStatement="evidenceStatement"
                itemCardViewModel={
                  Object {
                    "bankKey": 1,
                    "claimCode": "claim_1234",
                    "claimLabel": "claim_math",
                    "grade": 1,
                    "gradeLabel": "1",
                    "interactionTypeCode": "type01",
                    "interactionTypeLabel": "type01label",
                    "isPerformanceItem": false,
                    "itemKey": 123,
                    "subjectCode": "Math",
                    "subjectLabel": "MTH",
                    "targetHash": 323,
                    "targetShortName": "something",
                    "title": "test_title",
                  }
                }
                rubrics={Array []}
                targetDescription="targetDescription"
              />
            </div>
          </div>
        </div>,
      ],
      "className": "item-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changedTab": [Function],
          "selectedTab": "information",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <div
              className="item-content"
            >
              <div>
                <AboutThisItemDetail
                  commonCoreStandardsDescription="commonCoreStandardsDescription"
                  depthOfKnowledge="depthOfKnowledge"
                  educationalDifficulty="educationalDifficulty"
                  evidenceStatement="evidenceStatement"
                  itemCardViewModel={
                    Object {
                      "bankKey": 1,
                      "claimCode": "claim_1234",
                      "claimLabel": "claim_math",
                      "grade": 1,
                      "gradeLabel": "1",
                      "interactionTypeCode": "type01",
                      "interactionTypeLabel": "type01label",
                      "isPerformanceItem": false,
                      "itemKey": 123,
                      "subjectCode": "Math",
                      "subjectLabel": "MTH",
                      "targetHash": 323,
                      "targetShortName": "something",
                      "title": "test_title",
                    }
                  }
                  rubrics={Array []}
                  targetDescription="targetDescription"
                />
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <AboutThisItemDetail
                  commonCoreStandardsDescription="commonCoreStandardsDescription"
                  depthOfKnowledge="depthOfKnowledge"
                  educationalDifficulty="educationalDifficulty"
                  evidenceStatement="evidenceStatement"
                  itemCardViewModel={
                    Object {
                      "bankKey": 1,
                      "claimCode": "claim_1234",
                      "claimLabel": "claim_math",
                      "grade": 1,
                      "gradeLabel": "1",
                      "interactionTypeCode": "type01",
                      "interactionTypeLabel": "type01label",
                      "isPerformanceItem": false,
                      "itemKey": 123,
                      "subjectCode": "Math",
                      "subjectLabel": "MTH",
                      "targetHash": 323,
                      "targetShortName": "something",
                      "title": "test_title",
                    }
                  }
                  rubrics={Array []}
                  targetDescription="targetDescription"
                />
              </div>,
              "className": "item-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AboutThisItemDetail
                  commonCoreStandardsDescription="commonCoreStandardsDescription"
                  depthOfKnowledge="depthOfKnowledge"
                  educationalDifficulty="educationalDifficulty"
                  evidenceStatement="evidenceStatement"
                  itemCardViewModel={
                    Object {
                      "bankKey": 1,
                      "claimCode": "claim_1234",
                      "claimLabel": "claim_math",
                      "grade": 1,
                      "gradeLabel": "1",
                      "interactionTypeCode": "type01",
                      "interactionTypeLabel": "type01label",
                      "isPerformanceItem": false,
                      "itemKey": 123,
                      "subjectCode": "Math",
                      "subjectLabel": "MTH",
                      "targetHash": 323,
                      "targetShortName": "something",
                      "title": "test_title",
                    }
                  }
                  rubrics={Array []}
                  targetDescription="targetDescription"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "commonCoreStandardsDescription": "commonCoreStandardsDescription",
                  "depthOfKnowledge": "depthOfKnowledge",
                  "educationalDifficulty": "educationalDifficulty",
                  "evidenceStatement": "evidenceStatement",
                  "itemCardViewModel": Object {
                    "bankKey": 1,
                    "claimCode": "claim_1234",
                    "claimLabel": "claim_math",
                    "grade": 1,
                    "gradeLabel": "1",
                    "interactionTypeCode": "type01",
                    "interactionTypeLabel": "type01label",
                    "isPerformanceItem": false,
                    "itemKey": 123,
                    "subjectCode": "Math",
                    "subjectLabel": "MTH",
                    "targetHash": 323,
                    "targetShortName": "something",
                    "title": "test_title",
                  },
                  "rubrics": Array [],
                  "targetDescription": "targetDescription",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ItemTabs
            changedTab={[Function]}
            selectedTab="information"
          />,
          <div>
             
            <div
              className="item-content"
            >
              <div>
                <AboutThisItemDetail
                  commonCoreStandardsDescription="commonCoreStandardsDescription"
                  depthOfKnowledge="depthOfKnowledge"
                  educationalDifficulty="educationalDifficulty"
                  evidenceStatement="evidenceStatement"
                  itemCardViewModel={
                    Object {
                      "bankKey": 1,
                      "claimCode": "claim_1234",
                      "claimLabel": "claim_math",
                      "grade": 1,
                      "gradeLabel": "1",
                      "interactionTypeCode": "type01",
                      "interactionTypeLabel": "type01label",
                      "isPerformanceItem": false,
                      "itemKey": 123,
                      "subjectCode": "Math",
                      "subjectLabel": "MTH",
                      "targetHash": 323,
                      "targetShortName": "something",
                      "title": "test_title",
                    }
                  }
                  rubrics={Array []}
                  targetDescription="targetDescription"
                />
              </div>
            </div>
          </div>,
        ],
        "className": "item-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changedTab": [Function],
            "selectedTab": "information",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <div
                className="item-content"
              >
                <div>
                  <AboutThisItemDetail
                    commonCoreStandardsDescription="commonCoreStandardsDescription"
                    depthOfKnowledge="depthOfKnowledge"
                    educationalDifficulty="educationalDifficulty"
                    evidenceStatement="evidenceStatement"
                    itemCardViewModel={
                      Object {
                        "bankKey": 1,
                        "claimCode": "claim_1234",
                        "claimLabel": "claim_math",
                        "grade": 1,
                        "gradeLabel": "1",
                        "interactionTypeCode": "type01",
                        "interactionTypeLabel": "type01label",
                        "isPerformanceItem": false,
                        "itemKey": 123,
                        "subjectCode": "Math",
                        "subjectLabel": "MTH",
                        "targetHash": 323,
                        "targetShortName": "something",
                        "title": "test_title",
                      }
                    }
                    rubrics={Array []}
                    targetDescription="targetDescription"
                  />
                </div>
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div>
                  <AboutThisItemDetail
                    commonCoreStandardsDescription="commonCoreStandardsDescription"
                    depthOfKnowledge="depthOfKnowledge"
                    educationalDifficulty="educationalDifficulty"
                    evidenceStatement="evidenceStatement"
                    itemCardViewModel={
                      Object {
                        "bankKey": 1,
                        "claimCode": "claim_1234",
                        "claimLabel": "claim_math",
                        "grade": 1,
                        "gradeLabel": "1",
                        "interactionTypeCode": "type01",
                        "interactionTypeLabel": "type01label",
                        "isPerformanceItem": false,
                        "itemKey": 123,
                        "subjectCode": "Math",
                        "subjectLabel": "MTH",
                        "targetHash": 323,
                        "targetShortName": "something",
                        "title": "test_title",
                      }
                    }
                    rubrics={Array []}
                    targetDescription="targetDescription"
                  />
                </div>,
                "className": "item-content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <AboutThisItemDetail
                    commonCoreStandardsDescription="commonCoreStandardsDescription"
                    depthOfKnowledge="depthOfKnowledge"
                    educationalDifficulty="educationalDifficulty"
                    evidenceStatement="evidenceStatement"
                    itemCardViewModel={
                      Object {
                        "bankKey": 1,
                        "claimCode": "claim_1234",
                        "claimLabel": "claim_math",
                        "grade": 1,
                        "gradeLabel": "1",
                        "interactionTypeCode": "type01",
                        "interactionTypeLabel": "type01label",
                        "isPerformanceItem": false,
                        "itemKey": 123,
                        "subjectCode": "Math",
                        "subjectLabel": "MTH",
                        "targetHash": 323,
                        "targetShortName": "something",
                        "title": "test_title",
                      }
                    }
                    rubrics={Array []}
                    targetDescription="targetDescription"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "commonCoreStandardsDescription": "commonCoreStandardsDescription",
                    "depthOfKnowledge": "depthOfKnowledge",
                    "educationalDifficulty": "educationalDifficulty",
                    "evidenceStatement": "evidenceStatement",
                    "itemCardViewModel": Object {
                      "bankKey": 1,
                      "claimCode": "claim_1234",
                      "claimLabel": "claim_math",
                      "grade": 1,
                      "gradeLabel": "1",
                      "interactionTypeCode": "type01",
                      "interactionTypeLabel": "type01label",
                      "isPerformanceItem": false,
                      "itemKey": 123,
                      "subjectCode": "Math",
                      "subjectLabel": "MTH",
                      "targetHash": 323,
                      "targetShortName": "something",
                      "title": "test_title",
                    },
                    "rubrics": Array [],
                    "targetDescription": "targetDescription",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ItemCardViewer renders rubric tab 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemCardViewer
    item={
      Object {
        "commonCoreStandardsDescription": "commonCoreStandardsDescription",
        "depthOfKnowledge": "depthOfKnowledge",
        "educationalDifficulty": "educationalDifficulty",
        "evidenceStatement": "evidenceStatement",
        "itemCardViewModel": Object {
          "bankKey": 1,
          "claimCode": "claim_1234",
          "claimLabel": "claim_math",
          "grade": 1,
          "gradeLabel": "1",
          "interactionTypeCode": "type01",
          "interactionTypeLabel": "type01label",
          "isPerformanceItem": false,
          "itemKey": 123,
          "subjectCode": "Math",
          "subjectLabel": "MTH",
          "targetHash": 323,
          "targetShortName": "something",
          "title": "test_title",
        },
        "rubrics": Array [],
        "targetDescription": "targetDescription",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ItemTabs
          changedTab={[Function]}
          selectedTab="rubric"
        />,
        <div>
           
          <div
            className="item-content"
          />
           
        </div>,
      ],
      "className": "item-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changedTab": [Function],
          "selectedTab": "rubric",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <div
              className="item-content"
            />,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "item-content",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          " ",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ItemTabs
            changedTab={[Function]}
            selectedTab="rubric"
          />,
          <div>
             
            <div
              className="item-content"
            />
             
          </div>,
        ],
        "className": "item-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changedTab": [Function],
            "selectedTab": "rubric",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <div
                className="item-content"
              />,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "item-content",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            " ",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ItemCardViewer renders viewer tab 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemCardViewer
    item={
      Object {
        "commonCoreStandardsDescription": "commonCoreStandardsDescription",
        "depthOfKnowledge": "depthOfKnowledge",
        "educationalDifficulty": "educationalDifficulty",
        "evidenceStatement": "evidenceStatement",
        "itemCardViewModel": Object {
          "bankKey": 1,
          "claimCode": "claim_1234",
          "claimLabel": "claim_math",
          "grade": 1,
          "gradeLabel": "1",
          "interactionTypeCode": "type01",
          "interactionTypeLabel": "type01label",
          "isPerformanceItem": false,
          "itemKey": 123,
          "subjectCode": "Math",
          "subjectLabel": "MTH",
          "targetHash": 323,
          "targetShortName": "something",
          "title": "test_title",
        },
        "rubrics": Array [],
        "targetDescription": "targetDescription",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ItemTabs
          changedTab={[Function]}
          selectedTab="viewer"
        />,
        <div>
           
          <div
            className="item-content"
          >
            <ItemFrame
              url="http://ivs.smarterbalanced.org/items?ids=1-123"
            />
          </div>
           
        </div>,
      ],
      "className": "item-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changedTab": [Function],
          "selectedTab": "viewer",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            <div
              className="item-content"
            >
              <ItemFrame
                url="http://ivs.smarterbalanced.org/items?ids=1-123"
              />
            </div>,
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ItemFrame
                url="http://ivs.smarterbalanced.org/items?ids=1-123"
              />,
              "className": "item-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "url": "http://ivs.smarterbalanced.org/items?ids=1-123",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          " ",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ItemTabs
            changedTab={[Function]}
            selectedTab="viewer"
          />,
          <div>
             
            <div
              className="item-content"
            >
              <ItemFrame
                url="http://ivs.smarterbalanced.org/items?ids=1-123"
              />
            </div>
             
          </div>,
        ],
        "className": "item-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changedTab": [Function],
            "selectedTab": "viewer",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              <div
                className="item-content"
              >
                <ItemFrame
                  url="http://ivs.smarterbalanced.org/items?ids=1-123"
                />
              </div>,
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ItemFrame
                  url="http://ivs.smarterbalanced.org/items?ids=1-123"
                />,
                "className": "item-content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "url": "http://ivs.smarterbalanced.org/items?ids=1-123",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            " ",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
